examples/integrations/oci/atp-reactive/src/main/java/io/helidon/examples/integrations/oci/atp/reactive/AtpService.java

```diff
@@ -52,36 +57,53 @@ public void update(Routing.Rules rules) {
-                            }
-                            zipStream.close();
-                        } catch (Exception e) {
-                            LOGGER.log(Level.WARNING, "Exception while processing wallet content", e);
-                            res.status(Http.Status.INTERNAL_SERVER_ERROR_500).send();
-                        }
-                        res.status(Http.Status.OK_200).send();
+                .map(ApiOptionalResponse::entity)
+                .map(e -> e.map(GenerateAutonomousDatabaseWallet.Response::walletArchive))
+                .flatMap(maybeArchive ->
+                        maybeArchive.map(archive -> {
+                                    try {
+                                        return createDbClient(archive).execute(exec -> exec.query("SELECT 'Hello world!!' FROM DUAL"));
+                                    } catch (SQLException sqlException) {
+                                        LOGGER.log(Level.SEVERE, "Error creating DbClient", sqlException);
+                                    }
+                                    res.status(500).send();
+                                    return null;
+                                }
+                        ).orElseGet(() -> {
+                            res.status(404).send();
+                            return Multi.empty();
+                        })
+                )
+                .first()
+                .map(dbRow -> dbRow.column(1).as(String.class))
+                .onError(res::send)
+                .forSingle(res::send);
+    }
+
+    DbClient createDbClient(GenerateAutonomousDatabaseWallet.WalletArchive walletArchive) throws SQLException {
+        PoolDataSource pds = PoolDataSourceFactory.getPoolDataSource();
+        try {
+            pds.setSSLContext(walletArchive.getSSLContext());
+            pds.setURL(walletArchive.getJdbcUrl(config.get("db.serviceName")
+                    .as(String.class)
+                    .orElseThrow(() -> new IllegalStateException("Missing serviceName!!"))));
+            pds.setUser(config.get("db.userName").as(String.class).orElse("ADMIN"));
+            pds.setPassword(config.get("db.password")
+                    .as(String.class)
+                    .orElseThrow(() -> new IllegalStateException("Missing password!!")));
+            pds.setConnectionFactoryClassName(OracleDataSource.class.getName());
+        } catch (SQLException e) {
+            LOGGER.log(Level.SEVERE, "Error setting up PoolDataSource", e);
+            throw e;
+        }
+        return new JdbcDbClientProvider().builder()
+                .connectionPool(() -> {
+                    try {
+                        return pds.getConnection();
+                    } catch (SQLException e) {
+                        throw new IllegalStateException("Error while setting up new connection", e);
                     }
                 })
-                .exceptionally(res::send);
+                .build();
```

Comment:
Returning Single from `createDbClient` can help with those ugly checked exceptions
```java
Single<DbClient> createDbClient(...
try{
...
} catch (SQLException e) {
  LOGGER...
  return Single.error(e);
}
return Single.just(new JdbcDbClientProvider().builder()..connectionPool(() -> {...}).build());
```
Eg.:
```java
.flatMap(maybeArchive ->
    maybeArchive.map(archive -> 
          createDbClient(archive).execute(exec -> exec.query("SELECT 'Hello world!!' FROM DUAL"))
                    ).orElseGet(() -> {
                            res.status(404).send();
                            return Multi.empty();
                    })
        )
```

Reply:
that 404 is exactly the usecase for `ifEmpty` #3382, we will have it soon

Reply:
Thanks, updated code to be more "reactive".